### PR TITLE
KafkaSource: adding authentication support

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubali@redhat.com>

Fixes: #3

### CODE REVIEW

File: pkg/reconciler/kafka/kafka.go [hunk 1]
```diff
@@ -36,7 +39,7 @@ func GenerateScaleTargetName(src *kafkav1beta1.KafkaSource) string {
 	return eventingutils.GenerateFixedName(src, fmt.Sprintf("kafkasource-%s", src.Name))
 }
 
-func GenerateScaleTriggers(src *kafkav1beta1.KafkaSource) ([]kedav1alpha1.ScaleTriggers, error) {
+func GenerateScaleTriggers(src *kafkav1beta1.KafkaSource, triggerAuthentication *kedav1alpha1.TriggerAuthentication) ([]kedav1alpha1.ScaleTriggers, error) {
 	triggers := []kedav1alpha1.ScaleTriggers{}
 	bootstrapServers := strings.Join(src.Spec.BootstrapServers[:], ",")
```
Comment (lionelvillard) [inline pkg/reconciler/kafka/kafka.go:44]: why creating a slice here? Just `src.Spec.BootstrapServers` should work, right?

Comment (zroubalik) [reply pkg/reconciler/kafka/kafka.go:44]: `bootstrapServers` in `ScaleTrigger{}` is type of string

Comment (lionelvillard) [reply pkg/reconciler/kafka/kafka.go:44]: I meant the `[:]`

Comment (zroubalik) [reply pkg/reconciler/kafka/kafka.go:44]: Sorry, I don't follow 😅 Could you please send me the suggestion directly?

`src.SpecBootstrapServers` is `[]string ` but `bootstrapServers` in ScaleTrigger{} is type of `string`

Comment (zroubalik) [reply pkg/reconciler/kafka/kafka.go]: Fixed

**Pull request merged**